src/state_point.cpp

```diff
@@ -577,58 +565,59 @@ void write_source_point(const char* filename, bool surf_source_bank)
   bool parallel = false;
 #endif
 
-  std::string filename_;
-  if (filename) {
-    filename_ = filename;
-  } else {
-    // Determine width for zero padding
-    int w = std::to_string(settings::n_max_batches).size();
+  if (!filename)
+    fatal_error("write_source_point filename needs a nonempty name.");
 
-    filename_ = fmt::format("{0}source.{1:0{2}}.h5", settings::path_output,
-      simulation::current_batch, w);
+  std::string filename_(filename);
+  const auto extension = get_file_extension(filename_);
+  if (extension == "") {
+    filename_.append(".h5");
+  } else if (extension != "h5") {
+    fatal_error("openmc_source_point was passed an incorrect file "
+                "extension. Must either have no file extension or .h5");
```

Comment:
Same consideration here -- let's make this a warning. Also `openmc_source_point` is not a thing, so the wording should be updated as well